File: 2347-best-poker-hand/2347-best-poker-hand.java
class Solution {
    public String bestHand(int[] ranks, char[] suits) {
        HashMap<Integer, Integer> r = new HashMap<Integer, Integer>();
        HashMap<Character, Integer> s = new HashMap<Character, Integer>();
        for (char cur : suits) {
            s.put(cur, s.getOrDefault(cur, 0) + 1);
        }
        if (s.size() == 1) return "Flush";
        String res = "High Card";
        for (int cur : ranks) {
            if (r.containsKey(cur)) {
                r.put(cur, r.get(cur) + 1);
                if (r.get(cur) == 2) {
                    res = "Pair";
                } else if (r.get(cur) == 3) {
                    res = "Three of a Kind";
                    break;
                }
            } else {
                r.put(cur, 1);
            }
        }
        return res;
    }
}

File: 2347-best-poker-hand/2347-best-poker-hand.js
/**
 * @param {number[]} ranks
 * @param {character[]} suits
 * @return {string}
 */
var bestHand = function(ranks, suits) {
    if(suits.length===suits.filter((v,i)=>suits[0]===v).length) return "Flush"
    
    let dict={}
    for(let i=0;i<ranks.length;i++){
        dict[ranks[i]]?dict[ranks[i]]+=1:dict[ranks[i]]=1
    }
    let max=-1
    for(value of Object.values(dict))
    {
        if(value>max) max=value
    }
    if(max>=3) return "Three of a Kind"
    if(max===2) return "Pair"
    return "High Card"
    
    
};